### PR TITLE
Corrected incomplete error handling and validation of configuration

### PR DESCRIPTION
**Pull request by userlocalhost**
This is a patch for #22.

This patch
- corrected incomplete error handling and validation of configuration.
- added the test-case of initializing an 'Action' object with a variety of configuration.

Thank you.

### CODE REVIEW

File: actions/lib/action.py [hunk 1]
```diff
@@ -22,12 +22,14 @@ def __init__(self, config):
...
             'aws_secret_access_key': None
         }
+        self.userdata = None
 
-        if config['st2_user_data']:
-            with open(config['st2_user_data'], 'r') as fp:
-                self.userdata = fp.read()
-        else:
-            self.userdata = None
+        if 'st2_user_data' in config and config['st2_user_data']:
```
Comment (Kami) [inline actions/lib/action.py]: Short hand notation which should also work is `config.get('st2_user_data', None)`.

**Comment by Kami** [comment]
Nice work, thanks.

The change looks good to me, let's just please also add `aws.example.yaml` example config file in the pack root directory. This file servers as an example to users using this pack and in addition to that we also validate it against the schema to make sure it's valid on each CI run.

**Comment by Kami** [comment]
Another thing - I believe we also need to update `config.schema.yaml` and add this user data option there with a default value of None.

File: actions/lib/action.py (hunk 1 above)
Comment (dzimine) [reply actions/lib/action.py]: ... and `None` is default so `config.get('st2_user_data')` works.

**Review by dzimine — APPROVED** [review]

**Comment by userlocalhost** [comment]
Thank you for your review, @kami.

I wrote an example file of configuration for users.

> Another thing - I believe we also need to update `config.schema.yaml`

Sure. I also think so that. I'll make an another PR for that.

Thank you.

**Pull request merged**